airflow/www/fab_security/manager.py

```diff
@@ -85,6 +85,33 @@ def _oauth_tokengetter(token=None):
     return token
 
 
+class AnonymousUser(AnonymousUserMixin):
+    """User object used when no active user is logged in."""
+
+    _roles = set()
+    _perms = set()
+
+    @property
+    def roles(self):
+        if not self._roles:
+            public_role = current_app.appbuilder.get_app.config["AUTH_ROLE_PUBLIC"]
+            self._roles = {current_app.appbuilder.sm.find_role(public_role)} if public_role else set()
+        return self._roles
+
+    @roles.setter
+    def roles(self, roles):
+        self._roles = roles
+        self._perms = set()
+
+    @property
+    def perms(self):
+        if not self._perms:
+            self._perms = set()
+            for role in self.roles:
+                self._perms.update({(perm.action.name, perm.resource.name) for perm in role.permissions})
+        return self._perms
```

Comment:
`from airflow.compat.functools import cached_property` and then:

```suggestion
    @cached_property
    def perms(self):
        _perms = set()
        for role in self.roles:
            _perms.update({(perm.action.name, perm.resource.name) for perm in role.permissions})
        return _perms
```
And then `del self.perms` in the `@roles.setter` to clear the cache.

(This is all functionally the same, I just find the delcarative `@cached_property` clearer to read.)

Reply:
I'll make this change in a follow up PR